### PR TITLE
WFLY-5606 ManagementOnlyModeTestCase uses non-standard reload idioms

### PR DESCRIPTION
**Pull request by ctomc**
this is follow up from https://github.com/wildfly/wildfly/pull/8367

Test was changed to use WildFlyRunner and as such was moved to manual testsuite which is where it should be from begining.

### CODE REVIEW

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/management/api/SimpleSubsystemsTestCase.java [hunk 1]
```diff
@@ -40,6 +40,7 @@
  */
 @RunWith(Arquillian.class)
 @RunAsClient
+//todo move to wildfly-core
```
Comment (bstansberry) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/management/api/SimpleSubsystemsTestCase.java]: Why? AIUI it's just a trivial test of some subsystems in full.

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/management/api/expression/Utils.java [hunk 2]
```diff
@@ -44,20 +44,23 @@
     public static void setProperty(String name, String value, ModelControllerClient client) {
         ModelNode modelNode = createOpNode("system-property=" + name, ADD);
         modelNode.get(VALUE).set(value);
+        modelNode.get(OPERATION_HEADERS).get(ALLOW_RESOURCE_SERVICE_RESTART).set(true);
```
Comment (bstansberry) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/management/api/expression/Utils.java]: This looks like cruft. SystemPropertyAddHandler does not respect this header, and in any case adding the props used in the test shouldn't trigger reload-required. If they do there's a bug.

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/management/api/expression/Utils.java [hunk 3]
```diff
@@ -44,20 +44,23 @@
...
         ModelNode result = executeOp(modelNode, client);
         log.debugf("Added property %s, result: %s", name, result);
     }
     
     public static void removeProperty(String name, ModelControllerClient client) {
         ModelNode modelNode = createOpNode("system-property=" + name, REMOVE);
         ModelNode result = executeOp(modelNode, client);
+        modelNode.get(OPERATION_HEADERS).get(ALLOW_RESOURCE_SERVICE_RESTART).set(true);
```
Comment (bstansberry) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/management/api/expression/Utils.java]: This looks like cruft. SystemPropertyRemoveHandler does not respect this header, and in any case removing the props used in the test shouldn't trigger reload-required. If they do there's a bug.

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/management/api/expression/Utils.java [hunk 4]
```diff
@@ -44,20 +44,23 @@
...
         log.debugf("Removing property %s. Result: %s.", name, result);
     }
     
     public static void redefineProperty(String name, String value, ModelControllerClient client) {
         ModelNode modelNode = createOpNode("system-property=" + name, WRITE_ATTRIBUTE_OPERATION);
         modelNode.get(NAME).set(VALUE);
         modelNode.get(VALUE).set(value);
+        modelNode.get(OPERATION_HEADERS).get(ALLOW_RESOURCE_SERVICE_RESTART).set(true);
```
Comment (bstansberry) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/management/api/expression/Utils.java]: This looks like cruft. SystemPropertyValueWriteAttributeHandler does not respect this header, and in any case updating the props used in the test shouldn't trigger reload-required. If they do there's a bug.

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/management/api/expression/ExpressionSubstitutionInContainerTestCase.java [hunk 5]
```diff
@@ -208,44 +205,44 @@ public void testMultipleLevelExpression() {
...
      *    <property name="qa.test.property" value="${qa.test.exp:defaultValue}"/>
      * </system-properties>
-     * 
+     *
      * Write attribute set:
      * <property name="qa.test.exp" value="expression.value"/>
      */
-    @Ignore("AS7-6431")  // for this test works there will be :reload after redefinition
     @Test
+    @Ignore("AS7-6431")
```
Comment (bstansberry) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/management/api/expression/ExpressionSubstitutionInContainerTestCase.java]: Feel free to comment out (don't delete) the tests that are ignored due to AS7-6431. I've rejected that JIRA.

File: testsuite/integration/manualmode/src/test/java/org/jboss/as/test/manualmode/model/ReadFullModelTestCase.java [hunk 6]
```diff
@@ -26,63 +26,54 @@
...
-            ModelNode result = ModelTestUtils.checkResultAndGetContents(client.execute(rr));
+            ModelNode result = client.executeForResult(rr);
 
             //Just a quick sanity test to check that we are full-ha by making sure some of the expected subsystems are there
             Assert.assertTrue(result.hasDefined(SUBSYSTEM, "messaging-activemq"));
             Assert.assertTrue(result.hasDefined(SUBSYSTEM, "jgroups"));
-
-            ServerReload.executeReloadAndWaitForCompletion(client, true);
-            result = ModelTestUtils.checkResultAndGetContents(client.execute(rr));
+            client.executeForResult(rr);
```
Comment (bstansberry) [inline testsuite/integration/manualmode/src/test/java/org/jboss/as/test/manualmode/model/ReadFullModelTestCase.java]: This test no longer tests both normal and admin-only. Before it started normal and reloaded to admin only, now it starts admin-only, doesn't reload and the executes the read op again.

Comment (ctomc) [inline testsuite/integration/manualmode/src/test/java/org/jboss/as/test/manualmode/model/ReadFullModelTestCase.java]: ah, I misread what code was testing, as it was using the usual wrong idom of starting server and than reloading in admin mode to do its work. Will fix.

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/management/api/expression/Utils.java (hunk 4 above)
Comment (ctomc) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/management/api/expression/Utils.java]: you are right, not sure why I added that at the time. will remove.

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/management/api/SimpleSubsystemsTestCase.java (hunk 1 above)
Comment (ctomc) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/management/api/SimpleSubsystemsTestCase.java]: it only tests "simple" subsystems 
aka subsystems that have no resources or configuration options. so no need to have something like that in full. 
If we really need tests for something like that.

File: testsuite/integration/manualmode/src/test/java/org/jboss/as/test/manualmode/model/ReadFullModelTestCase.java [hunk 7]
```diff
@@ -26,63 +26,55 @@
...
  */
-@RunWith(Arquillian.class)
-@RunAsClient
+@ServerControl(manual = true)
+@RunWith(WildflyTestRunner.class)
 public class ReadFullModelTestCase {
 
     //This is the full-ha setup which is the fullest config we have
     //Reuse this rather than a setup
-    public static final String SERVER = "jbossas-messaging-ha-server1";
```
Comment (ctomc) [inline testsuite/integration/manualmode/src/test/java/org/jboss/as/test/manualmode/model/ReadFullModelTestCase.java]: this was actually wrong before as it only started server with this name (in our target/<name-of-server>) folder not with standalone-full-ha.xml configuration.

**Comment by pferraro** [comment]
That last test failure is fixed by https://github.com/wildfly/wildfly/pull/9134

**Pull request merged**